zerver/actions/custom_profile_fields.py

```diff
@@ -209,6 +216,48 @@ def do_update_user_custom_profile_data_if_changed(
             },
         )
 
+        new_value = get_custom_profile_field_display_value(field_value)
+
+        changes.append(
+            UserProfileChangeDict(
+                field_name=field_value.field.name,
+                old_value=old_value,
+                new_value=new_value,
+            )
+        )
+
+    if changes and notify:
+        send_user_profile_update_notification(
+            user_profile=user_profile, acting_user=acting_user, changes=changes
+        )
+
+
+def get_custom_profile_field_display_value(field_value: CustomProfileFieldValue) -> str:
+    """Convert custom profile field value to human-readable string based on field type."""
+    type = field_value.field.field_type
+
+    if type == CustomProfileField.SELECT:
+        field_data_dict = orjson.loads(field_value.field.field_data)
+        value_key = field_value.value
+        if value_key in field_data_dict:
+            return field_data_dict[value_key]["text"]
+        else:
+            return field_value.value
+
+    elif type == CustomProfileField.USER:
+        if not field_value.value:
+            return ""
+        try:
+            user_ids = orjson.loads(field_value.value)
+            users_names = UserProfile.objects.filter(id__in=user_ids).values_list(
+                "full_name", flat=True
+            )
+            return (", ").join(users_names)
+        except (orjson.JSONDecodeError, TypeError):
+            return field_value.value
+    else:
+        return field_value.value
```

Comment:
Should we show `rendered_value` in the notification messages for renderable fields?

Reply:
Yes, the current approach is fine.